### PR TITLE
basic spam detection

### PR DESCRIPTION
**Pull request by sgfost**
uses two methods:
- timing form load->submission time
- using a honeypot field hidden from real users with css

logs actions to a SpamModeration model with a generic fk to content models. actions such as deactivating the user or allowing the post can then be taken via a WIP admin view

#### TODO:
- [x] test coverage
- [x] usable admin view with
  - confirming/rejecting spam assessment
  - filtering
  - option to deactivate user on confirm (or default?)
  - email reminder when X number of spam objects pile up, or after Y amt of time (?)
- [x] add more context to `SpamContent` detailing how/why we think its spam, flexible enough that it can also be used for detecting with NLP (think ahead to reconciling #693)

resolves comses/planning#235

### CODE REVIEW

**Comment by sgfost** [comment]
spam content, and deleted events/jobs for that matter, still show up in the site-wide search. Need to find a good way to filter those from the general search but struggling to figure out how it currently does this for non-live codebases..

**Comment by alee** [comment]
> spam content, and deleted events/jobs for that matter, still show up in the site-wide search. Need to find a good way to filter those from the general search but struggling to figure out how it currently does this for non-live codebases..

good catch, we should have a 
```
@classmethod
get_indexed_objects(cls)
```
defined for Events, MemberProfiles, Jobs

**Comment by sgfost** [comment]
ah that's it, thanks

The bulk of this is done, just need to put up a few tests. I think it ought to work as a more flexible base for using the work in #693 as another detection method, but lmk if you have any thoughts or concerns when you get around to it

File: django/core/views.py
Comment (alee) [inline django/core/views.py:456]: am I understanding this correctly to allow possible spam events to be accessible in the detail view? (same for Jobs?)

File: django/curator/views.py
Comment (alee) [inline django/curator/views.py:7]: should eventually migrate to Snippets or alternative https://github.com/comses/planning/issues/248

File: django/library/jinja2/library/codebases/releases/edit.jinja
Comment (alee) [inline django/library/jinja2/library/codebases/releases/edit.jinja]: Assume this should be `alert_if_spam(true)` ? and prevent release editing if marked as spam?

**Review by alee — APPROVED** [review]
LGTM thanks @sgfost ! I think I fixed a few minor issues with the codebase release edit form still including an outdated reference to the spam alert macro, will wait and see if things still look OK on staging before merging

File: django/core/jinja2/core/events/retrieve.jinja
Comment (sgfost) [reply django/core/jinja2/core/events/retrieve.jinja:25]: Yeah, its now treated exactly the same as things marked spam, hidden from list view but accessible if you have a direct link to it. There wasn't much reasoning behind it besides simplifying queries

File: django/core/views.py
Comment (sgfost) [reply django/core/views.py:456]: Yes, mostly to give an explanation of what happened should a real post get mistakenly flagged. Though it should probably only be accessible by admins and the submitter

File: django/core/mixins.py
Comment (sgfost) [reply django/core/mixins.py:189]: Probably, yeah. the mixin shouldn't be a serializer. I'll check this out

**Comment by sgfost** [comment]
Thx, I'll make a new issue with these needed changes as to not block merging if its in a 'good enough' state

**Pull request merged**
